### PR TITLE
:sparkles: Replace windup with analyzer-lsp

### LINKED ISSUE #244: :seedling: Replace the windup addon with the analyzer-lsp.
I think s/windup/analyzer/g should do it.

Also, now that the **hub** is storing analysis data in the DB, we should at least double the default DB volume size of 5G to 10G (or more).  Perhaps 50G.
For 1000 applications, 100 issues, 1500 incidents each, the tackle.db is 4.3G.  I have been using these numbers as a _medium_ sized deployment.
Now that we are no longer storing windup reports in the _bucket_ volume I think we can safely reduce this default of 100G a lot.  Perhaps down to 25G.
Also, given the MUCH more complex queries on large datasets, we should consider bumping the cpu & memory request/limit on the Hub.
```
hub_container_requests_cpu: "100m"
hub_container_limits_cpu: "500m"
hub_container_requests_memory: "350Mi"
hub_container_limits_memory: "800Mi"
```
should be probably be.
```
hub_container_requests_cpu: "500m"
hub_container_limits_cpu: "1"
hub_container_requests_memory: "1Gi"
hub_container_limits_memory: "2Gi"
```
I will try to get better benchmarks.

We may want to have a broader discussion on this. What is our target out-of-the-box deployment?
100 applications on a labtop w/ minikube?
Bigger?  Smaller?

### PR DESCRIPTION
**Pull request by djzager**
Fixes #244

### CODE REVIEW

**Comment by djzager** [comment]
Looks like the pod starts:

```
tackle2-hub on  main [!] via 🐹 v1.20.5
➜  kubectl describe -n konveyor-tackle pod/task-1-4jzz6
Name:             task-1-4jzz6
Namespace:        konveyor-tackle
Priority:         0
Service Account:  tackle-hub
Node:             minikube/192.168.49.2
Start Time:       Mon, 10 Jul 2023 13:50:21 -0400
Labels:           app=tackle
                  role=task
                  task=1
Annotations:      <none>
Status:           Pending
IP:
IPs:              <none>
Containers:
  main:
    Container ID:
    Image:          quay.io/konveyor/tackle2-addon-analyzer:latest
    Image ID:
    Port:           <none>
    Host Port:      <none>
    State:          Waiting
      Reason:       ContainerCreating
    Ready:          False
    Restart Count:  0
    Limits:
      cpu:     1
      memory:  6Gi
    Requests:
      cpu:     1
      memory:  4Gi
    Environment:
      HUB_BASE_URL:  http://tackle-hub.konveyor-tackle.svc:8080
      TASK:          1
      TOKEN:         <set to the key 'TOKEN' in secret 'task-1-ch87g'>  Optional: false
    Mounts:
      /cache from cache (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-7rfxt (ro)
Conditions:
  Type              Status
  Initialized       True
  Ready             False
  ContainersReady   False
  PodScheduled      True
Volumes:
  cache:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:
    SizeLimit:  <unset>
  kube-api-access-7rfxt:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   Burstable
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type    Reason     Age   From               Message
  ----    ------     ----  ----               -------
  Normal  Scheduled  26s   default-scheduler  Successfully assigned konveyor-tackle/task-1-4jzz6 to minikube
  Normal  Pulling    25s   kubelet            Pulling image "quay.io/konveyor/tackle2-addon-analyzer:latest"
```

Not sure if there is anything more to do at this point to verify it's working from the operator's perspective.

File: roles/tackle/defaults/main.yml [hunk 1]
```diff
@@ -24,17 +24,17 @@ hub_deployment_name: "{{ hub_service_name }}"
 hub_deployment_replicas: "1"
 hub_deployment_strategy: "Recreate"
 hub_container_name: "{{ hub_service_name }}"
-hub_container_limits_cpu: "500m"
-hub_container_limits_memory: "800Mi"
-hub_container_requests_cpu: "100m"
-hub_container_requests_memory: "350Mi"
+hub_container_limits_cpu: "1"
```
Comment (pranavgaikwad) [inline roles/tackle/defaults/main.yml:27]: @djzager do we really need limits?

**Review by pranavgaikwad — APPROVED** [review]
LGTM, I'd argue we dont need limits by default. Dont feel too strongly though.

**Comment by djzager** [comment]
```
tackle2-hub on  main [!] via 🐹 v1.20.5
➜  curl -X GET $(minikube ip)/hub/tasks/1 | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  2466    0  2466    0     0  1026k      0 --:--:-- --:--:-- --:--:-- 1204k
{
  "id": 1,
  "createUser": "",
  "updateUser": "",
  "createTime": "2023-07-10T17:50:20.621875127Z",
  "name": "Dog.1.windup",
  "addon": "analyzer",
  "data": {
    "mode": {
      "artifact": "",
      "binary": false,
      "diva": false,
      "withDeps": false
    },
    "rules": {
      "labels": {
        "excluded": [],
        "included": [
          "konveyor.io/target=cloud-readiness"
        ]
      },
      "path": "",
      "rulesets": [
        {
          "id": 2,
          "name": "Containerization"
        }
      ],
      "tags": {
        "excluded": []
      }
    },
    "scope": {
      "packages": {
        "excluded": [],
        "included": []
      },
      "withKnown": false
    },
    "sources": [],
    "tagger": {
      "enabled": true
    },
    "targets": []
  },
  "application": {
    "id": 1,
    "name": "Dog"
  },
  "state": "Failed",
  "image": "quay.io/konveyor/tackle2-addon-analyzer:latest",
  "bucket": {
    "id": 2
  },
  "started": "2023-07-10T18:13:17.519987938Z",
  "terminated": "2023-07-10T18:13:25.585307201Z",
  "errors": [
    {
      "severity": "Error",
      "description": "Pod failed: "
    }
  ],
  "pod": "konveyor-tackle/task-1-464gr",
  "retries": 1,
  "report": {
    "id": 1,
    "createUser": "admin.noauth",
    "updateUser": "admin.noauth",
    "createTime": "2023-07-10T18:13:19.320463815Z",
    "status": "Failed",
    "errors": [
      {
        "severity": "Error",
        "description": "[CMD] /usr/bin/konveyor-analyzer failed: exit status 1."
      }
    ],
    "activity": [
      "Fetching application.",
      "[CMD] Running: /usr/bin/ssh-agent -a /tmp/agent.1",
      "[CMD] succeeded.",
      "[SSH] Agent started.",
      "[GIT] Cloning: https://github.com/WASdev/sample.daytrader7.git",
      "[FILE] Created /addon/.gitconfig.",
      "[CMD] Running: /usr/bin/git clone https://github.com/WASdev/sample.daytrader7.git /addon/source/sample",
      "[CMD] succeeded.",
      "Settings: /addon/opt/settings.json",
      "> - name: go",
      ">   binaryPath: /usr/bin/golang-external-provider",
      ">   initConfig:",
      ">     - location: /addon/source/sample",
      ">       lspServerPath: /root/go/bin/gopls",
      ">       analysisMode: full",
      "> - name: java",
      ">   binaryPath: /jdtls/bin/jdtls",
      ">   initConfig:",
      ">     - location: /addon/source/sample",
      ">       lspServerPath: /jdtls/bin/jdtls",
      ">       analysisMode: source-only",
      ">       providerSpecificConfig:",
      ">         bundles: /jdtls/java-analyzer-bundle/java-analyzer-bundle.core/target/java-analyzer-bundle.core-1.0.0-SNAPSHOT.jar",
      "> - name: builtin",
      ">   initConfig:",
      ">     - location: /addon/source/sample",
      ">       analysisMode: \"\"",
      "[CMD] Running: /usr/bin/konveyor-analyzer --provider-settings /addon/opt/settings.json --output-file /addon/report.yaml --label-selector konveyor.io/target=cloud-readiness",
      "[CMD] failed: exit status 1.",
      "> unable to find rule path or file",
      "> "
    ],
    "task": 1
  }
}
```

File: roles/tackle/defaults/main.yml [hunk 2]
```diff
@@ -24,17 +24,17 @@ hub_deployment_name: "{{ hub_service_name }}"
...
+hub_container_requests_memory: "1Gi"
 hub_database_filename: "hub.db"
 hub_database_volume_name: "{{ hub_service_name }}-database"
-hub_database_volume_size: "5Gi"
+hub_database_volume_size: "10Gi"
 hub_database_volume_path: "/database"
 hub_database_volume_claim_name: "{{ hub_service_name }}-database-volume-claim"
 hub_bucket_volume_name: "{{ hub_service_name }}-bucket"
-hub_bucket_volume_size: "100Gi"
+hub_bucket_volume_size: "25Gi"
```
Comment (jortel) [inline roles/tackle/defaults/main.yml:37]: Recent conversations re: static analysis reports may still require upload of large files stored in buckets.  Not totally convinced we know enough to reduce this.

File: roles/tackle/defaults/main.yml (hunk 1 above)
Comment (jortel) [reply roles/tackle/defaults/main.yml:27]: Seems like we should have a limit to be a good citizen on the cluster.  That said, I don't know if 1 core is appropriate for out-of-the-box but felt sure that 100m was not enough.

**Comment by jwmatthews** [comment]
Fixes https://github.com/konveyor/enhancements/issues/108

**Review by jortel — APPROVED** [review]
LGTM

**Pull request merged**